client/src/main/kotlin/no/nav/su/se/bakover/client/oppdrag/avstemming/DetaljBuilder.kt

```diff
@@ -22,20 +21,20 @@ class DetaljBuilder(
         }
 
     private fun mapStatus(utbetaling: Utbetaling): Detaljdata.Detaljtype = when (utbetaling.kvittertMedFeilEllerVarsel()) {
-        true -> mapUtbetalingsstatus(utbetaling.kvittering!!.utbetalingsstatus)
+        true -> mapUtbetalingsstatus((utbetaling as Utbetaling.KvittertUtbetaling).kvittering.utbetalingsstatus)
         false -> Detaljdata.Detaljtype.MANGLENDE_KVITTERING
     }
 
     private fun mapUtbetalingsstatus(utbetalingsstatus: Utbetalingsstatus) = when (utbetalingsstatus) {
         OK_MED_VARSEL -> Detaljdata.Detaljtype.GODKJENT_MED_VARSEL
         FEIL -> Detaljdata.Detaljtype.AVVIST
-        else -> throw IllegalArgumentException("Funksjonell feil - skal ikke lage detajl for utbetalinger med status:$OK")
+        else -> throw IllegalArgumentException("Funksjonell feil - skal ikke lage detajl for utbetalinger med status:$utbetalingsstatus")
     }
 
     private fun shouldCreate(utbetaling: Utbetaling) =
         utbetaling.oversendtUtenKvittering() || utbetaling.kvittertMedFeilEllerVarsel()
 
-    private fun Utbetaling.oversendtUtenKvittering() = erOversendt() && !erKvittert()
+    private fun Utbetaling.oversendtUtenKvittering() = this is Utbetaling.OversendtUtbetaling && this.oppdragsmelding.erSendt()
```

Comment:
Hvorfor trenger vi begge? Det hadde gitt mer mening med ` this is Utbetaling.OversendtUtbetaling && !isKvittert()

Reply:
Jeg tror faktisk dette bare skyldes testdata. Etter refaktoreringen eksisterer det ikke lenger noe konsept som dekker "forsøkt sendte men feilede utbetalinger" ettersom utbetalingene kun vil persisteres dersom oversendelsen går bra.